queries.sql

```diff
@@ -2,192 +2,306 @@
+
 SELECT COUNT(DISTINCT a.name)
 FROM visits AS vs
-  JOIN vets AS vt ON vs.vet_id = vt.id
-  JOIN animals AS a ON vs.animal_id = a.id
+JOIN vets AS vt ON vs.vet_id = vt.id
+JOIN animals AS a ON vs.animal_id = a.id
 WHERE vt.name = 'Stephanie Mendez';
+
+
 SELECT vt.name AS vet_name,
-  sp.name AS species_name
+       sp.name AS species_name
 FROM specializations AS sz
-  JOIN species AS sp ON sz.species_id = sp.id
-  RIGHT JOIN vets AS vt ON sz.vet_id = vt.id
+JOIN species AS sp ON sz.species_id = sp.id
+RIGHT JOIN vets AS vt ON sz.vet_id = vt.id
 ORDER BY vt.name ASC;
+
+
 SELECT a.name AS animal_visited,
-  vs.visit_date
+       vs.visit_date
 FROM visits AS vs
-  JOIN animals AS a ON vs.animal_id = a.id
-  JOIN vets AS vt ON vs.vet_id = vt.id
+JOIN animals AS a ON vs.animal_id = a.id
+JOIN vets AS vt ON vs.vet_id = vt.id
 WHERE vt.name = 'Stephanie Mendez'
   AND vs.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+
 SELECT a.name AS animal_name,
-  COUNT(vs.*) AS number_of_visits
+       COUNT(vs.*) AS number_of_visits
 FROM visits AS vs
-  JOIN animals AS a ON vs.animal_id = a.id
+JOIN animals AS a ON vs.animal_id = a.id
 GROUP BY a.name
 ORDER BY number_of_visits DESC
 LIMIT 1;
+
+
 SELECT a.name AS animal_name
 FROM visits AS vs
-  JOIN vets AS vt ON vs.vet_id = vt.id
-  JOIN animals AS a ON vs.animal_id = a.id
+JOIN vets AS vt ON vs.vet_id = vt.id
+JOIN animals AS a ON vs.animal_id = a.id
 WHERE vt.name = 'Maisy Smith'
 ORDER BY vs.visit_date ASC
 LIMIT 1;
+
+
 SELECT a.name AS animal_name,
-  vt.name AS vet_name,
-  vs.visit_date
+       vt.name AS vet_name,
+       vs.visit_date
 FROM visits AS vs
-  JOIN vets AS vt ON vs.vet_id = vt.id
-  JOIN animals AS a ON vs.animal_id = a.id
+JOIN vets AS vt ON vs.vet_id = vt.id
+JOIN animals AS a ON vs.animal_id = a.id
 ORDER BY vs.visit_date DESC
 LIMIT 1;
+
+
 SELECT COUNT(*) AS num_visits
 FROM visits vs
-  JOIN animals a ON vs.animal_id = a.id
-  JOIN vets vt ON vs.vet_id = vt.id
-  LEFT JOIN specializations sp ON vt.id = sp.vet_id
-  AND a.species_id = sp.species_id
+JOIN animals a ON vs.animal_id = a.id
+JOIN vets vt ON vs.vet_id = vt.id
+LEFT JOIN specializations sp ON vt.id = sp.vet_id
+AND a.species_id = sp.species_id
 WHERE sp.vet_id IS NULL;
+
+
 SELECT sp.name AS specialization,
-  COUNT(sp.name)
+       COUNT(sp.name)
 FROM animals AS a
-  JOIN species AS sp ON a.species_id = sp.id
-  JOIN visits AS visit ON a.id = visit.animal_id
-  JOIN vets AS v ON v.id = visit.vet_id
+JOIN species AS sp ON a.species_id = sp.id
+JOIN visits AS visit ON a.id = visit.animal_id
+JOIN vets AS v ON v.id = visit.vet_id
 WHERE v.name = 'Maisy Smith'
 GROUP BY sp.name
 ORDER BY COUNT(sp.name) DESC
-LIMIT 1;
\ No newline at end of file
+LIMIT 1;
+
+/* Pull Request no.5 */ EXPLAIN ANALYZE
+SELECT COUNT(*)
+FROM visits
+where animal_id = 4;
+
+EXPLAIN ANALYZE
+SELECT *
+FROM visits
+where vet_id = 2;
+
+-- Expplain and analize
+EXPLAIN ANALYZE
+SELECT *
+FROM owners
+where email = 'owner_18327@mail.com';
```

Comment:
- [ ] The execution time for the two queries highlighted below is below **1000ms (1 second)**. 

```sql
EXPLAIN ANALYZE
SELECT COUNT(*)
FROM visits
where animal_id = 4;

EXPLAIN ANALYZE
SELECT *
FROM owners
where email = 'owner_18327@mail.com';
```

According to the requirements, the execution time of each of these queries should be above **1000ms (1 second)**. I would advise you to run these queries one or two more times to ensure there are many entries in the table before you run the query 👍 😎 :

```sql
INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
```

#### Hint: Please update your screenshots after implementing this. 👍

Reply:
Implemented!